_test/recurse3.go

```diff
@@ -0,0 +1,25 @@
+package main
+
+type F func(a *A)
+
+type A struct {
+	B string
+	D
+}
+
+type D struct {
+	*A
+	E *A
+	f F
+}
+
+func f1(a *A) { println("in f1", a.B) }
+
+func main() {
+	a := &A{B: "b"}
+	a.D = D{f: f1}
```

Comment:
Is this test actually proving anything about recursion? Since we're not setting/using any of the A fields of the D struct, I don't think it does.
it only shows that the A->D->F indirection works (which is not recursive afaics).

Well, ok, unless you count the fact that F takes an A as an argument (which is the same case as in recurse1.go I think).